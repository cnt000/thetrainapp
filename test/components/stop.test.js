import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Stop from '../../src/components/stop'

const mockStop = {
  location: {
    crs: 'WAT'
  },
  arrival: {
    notApplicable: true
  },
  departure: {
    scheduled: {
      scheduledTime: '2017-12-08T22:36:00+00:00',
      scheduledPlatform: '3'
    },
    realTime: {
      realTimeServiceInfo: {
        hasDeparted: true,
        realTime: '2017-12-08T22:37:00+00:00',
        realTimeFlag: 'Actual'
      }
    }
  },
  callingType: 'PickUp'
}

describe('Stop', () => {
  it('renders without crashing middle station', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <Stop stop={mockStop} isFirst={false} isLast={false} />,
      div
    )
  })

  test('has a valid snapshot middle station', () => {
    const component = renderer.create(
      <Stop stop={mockStop} isFirst={false} isLast={false} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('has a valid snapshot start station', () => {
    const component = renderer.create(
      <Stop stop={mockStop} isFirst isLast={false} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('has a valid snapshot end station', () => {
    const component = renderer.create(
      <Stop stop={mockStop} isFirst={false} isLast />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
