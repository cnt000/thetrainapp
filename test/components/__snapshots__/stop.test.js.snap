// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stop has a valid snapshot end station 1`] = `
<li
  className="sc-bdVaJa jIXPTc"
>
  <div
    className="sc-ifAKCX hZphQU"
  >
    <div
      className="sc-bxivhb kDOEdb"
    >
      23:36
    </div>
    <div
      className="sc-bwzfXH PgDNZ"
    />
    <div
      className="sc-htpNat hNGSJA"
    />
    <div
      className="sc-EHOje exgsEF"
    >
      <div
        className="calling-point-station"
      >
        WAT
      </div>
      <div
        className="calling-point-due"
      >
        Late 23:37
        <div
          className="sc-bZQynM gaKPBw"
        >
          departed
        </div>
      </div>
    </div>
  </div>
</li>
`;

exports[`Stop has a valid snapshot middle station 1`] = `
<li
  className="sc-bdVaJa jIXPTc"
>
  <div
    className="sc-ifAKCX hZphQU"
  >
    <div
      className="sc-bxivhb kDOEdb"
    >
      23:36
    </div>
    <div
      className="sc-bwzfXH PgDNZ"
    />
    <div
      className="sc-htpNat ipBCiG"
    />
    <div
      className="sc-EHOje exgsEF"
    >
      <div
        className="calling-point-station"
      >
        WAT
      </div>
      <div
        className="calling-point-due"
      >
        Late 23:37
        <div
          className="sc-bZQynM gaKPBw"
        >
          departed
        </div>
      </div>
    </div>
  </div>
</li>
`;

exports[`Stop has a valid snapshot start station 1`] = `
<li
  className="sc-bdVaJa jIXPTc"
>
  <div
    className="sc-ifAKCX hZphQU"
  >
    <div
      className="sc-bxivhb kDOEdb"
    >
      23:36
    </div>
    <div
      className="sc-bwzfXH PgDNZ"
    />
    <div
      className="sc-htpNat cWPbSB"
    />
    <div
      className="sc-EHOje exgsEF"
    >
      <div
        className="calling-point-station"
      >
        WAT
      </div>
      <div
        className="calling-point-due"
      >
        Late 23:37
        <div
          className="sc-bZQynM gaKPBw"
        >
          departed
        </div>
      </div>
    </div>
  </div>
</li>
`;
